### PR TITLE
fix: setter getter bug in multiple extends

### PR DESCRIPTION
**Pull request by JimmyDaddy**

### CODE REVIEW

File: lib/bone.js [hunk 1]
```diff
@@ -47,6 +48,8 @@ function compare(attributes, columns) {
   return diff;
 }
 
+const privateKeys = ['raw', 'rawSaved', 'rawUnset', 'rawPrevious', 'isNewRecord'];
```
Comment (JimmyDaddy) [inline lib/bone.js:51]: 将这几个 私有 key 转为 Symbol 的方式处理会不会好一点

File: lib/utils/index.js [hunk 2]
```diff
@@ -11,7 +11,27 @@ function compose() {
   return funcs.reverse().reduce((a, b) => (...arg) => b(a(...arg)));
 }
 
+function getPropertyNames (obj) {
+  if (obj == null) return [];
+  const propertyNames = [];
+  // avoid to deep clone obj
+  let muteObj = Object.getPrototypeOf(obj);
+  do {
+    propertyNames.push.apply(propertyNames, Object.getOwnPropertyNames(muteObj));
```
Comment (cyjake) [inline lib/utils/index.js:20]:
```suggestion
    propertyNames.push(...Object.getOwnPropertyNames(muteObj))
```

File: lib/utils/index.js [hunk 3]
```diff
@@ -11,7 +11,27 @@ function compose() {
...
+    muteObj = Object.getPrototypeOf(muteObj);
+  } while (muteObj);
```
Comment (cyjake) [inline lib/utils/index.js:22]: We can stop early if `muteObj.constructor.name === 'Bone'` or `muteObj.constructor.attributes` exists

File: lib/utils/index.js [hunk 4]
```diff
@@ -11,7 +11,27 @@ function compose() {
...
+  muteObj = {};
+  // get own properties
+  propertyNames.push.apply(propertyNames, Object.getOwnPropertyNames(obj));
+  // get unique properties' names, filter __proto__
+  for (const name of propertyNames) {
+    muteObj[name] = 1;
+  }
```
Comment (cyjake) [inline lib/utils/index.js:29]: How about use `Set()` in the first place? Could be handier than filtering with object key.

File: test/unit/adapters/sequelize.test.js [hunk 5]
```diff
@@ -836,6 +837,12 @@ describe('model.init with getterMethods and setterMethods', () => {
     }
   });
 
+  // multiple implement
+  class UU extends User {
+    get j () {
+      return 'j';
+    }
+  }
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:845]: `class CustomUser` might be preferred. If this fixture is only used in the `toJSON()` test case, should've considered putting it within that test. Or better create another specific test describing intended test purpose.

File: lib/bone.js [hunk 6]
```diff
@@ -91,18 +98,30 @@ class Bone {
...
+    }, {
+      key: rawPrevious,
+      value: {},
+    }, {
+      key: isNewRecord,
       value: true,
-      enumerable: false,
-      writable: true,
-      configurable: true,
+    }];
```
Comment (cyjake) [inline lib/bone.js:117]:
```js
Object.assign(this, {
  [raw]: {},
  [rawSaved]: {},
  [rawUnset]: new Set(),
  [rawPrevious]: {},
  isNewRecord: true,
});
```

File: lib/bone.js [hunk 7]
```diff
@@ -144,17 +163,57 @@ class Bone {
...
+    this[rawPrevious][key] = value;
+  }
+
+  isNewRecord() {
+    return this[isNewRecord];
+  }
+
+  setIsNewRecord(value) {
+    this[isNewRecord] = value;
+  }
```
Comment (cyjake) [inline lib/bone.js:214]: just a second thought, is turning `isNewRecord` into symbol really necessary?

Comment (JimmyDaddy) [reply lib/bone.js:214]: if not so it will be serialized while calling `toJSON` or `toObject`

**Pull request merged**